pkg/storage/db.go

```diff
@@ -230,16 +257,25 @@ func (db *DB) GeneratePartition(ctx context.Context) error {
 		return handleRollback(tx, err)
 	}
 
+	beginTime, err := time.Parse(time.RFC3339, begin)
+	if err != nil {
+		return handleRollback(tx, fmt.Errorf("Invalid partition range: %s, %v", latest, err))
+	}
 	latestTime, err := time.Parse(time.RFC3339, latest)
 	if err != nil {
 		return handleRollback(tx, fmt.Errorf("Invalid partition range: %s, %v", latest, err))
 	}
 
+	if beginTime.After(db.now()) {
```

Comment:
Nit: I'm finding the partition creation flow here a little hard to follow, I think adding some more comments throughout the code about how this should function might help make this easier to follow and reason about.

Reply:
I put some comments below, which did not update GH's comment ^^^.  /shrug